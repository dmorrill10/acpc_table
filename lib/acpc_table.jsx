import React from 'react';
import {CommunityInformation} from './community_information';
import {_, User, Opponent} from './player';

class AcpcTable extends React.Component {
  communityInformation() {
    return (<CommunityInformation
      key='community-information'
      cards={this.props.boardCards}
      potChips={this.props.potChips}/>);
  }
  user() {
    return (<User
      key='user'
      name={this.props.user.name}
      hole-cards={this.props.user.holeCards}
      dealer={this.props.user.dealer}
      acting={this.props.user.acting}
      chip-stack-amount={this.props.user.chipStackAmount}
      winnings={this.props.user.winnings}
      contribution={this.props.user.contribution}/>);
  }
  opponents() {
    let o = [];
    for (let i = 0; i < this.props.opponents.length; ++i) {
      let opponent = this.props.opponents[i];
      o.push(
        <li key={i}><Opponent
          name={opponent.name}
          hole-cards={opponent.holeCards}
          dealer={opponent.dealer}
          acting={opponent.acting}
          chip-stack-amount={opponent.chipStackAmount}
          winnings={opponent.winnings}
          contribution={opponent.contribution}/></li>
      );
    }
    return <ol key='opponents' className='opponents'>{o}</ol>;
  }
  render() {
    let l = [this.opponents(), this.communityInformation(), this.user()];
    return (
      <div className='acpc-table'>{l}</div>
    );
  }
}

export {AcpcTable};