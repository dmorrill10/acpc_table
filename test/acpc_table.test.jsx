import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {AcpcTable} from '../dist/acpc_table.js';

describe('AcpcTable', () => {
  describe("with 2-player Texas hold'em table on the river", () => {
    it('works', () => {
      const boardCards = [
        {
          rank: '2',
          suit: 'h'
        }, {
          rank: '3',
          suit: 'd'
        }, {
          rank: 'T',
          suit: 's'
        }, {
          rank: 'A',
          suit: 'c'
        }, {
          rank: 'K',
          suit: 'd'
        }
      ];
      const user = {
        name: 'player1',
        holeCards: [
          {}, {}
        ],
        dealer: true,
        acting: true,
        chipStackAmount: 1000,
        winnings: 0,
        contribution: 400
      };
      const opponents = [
        {
          name: 'player2',
          holeCards: [
            {}, {}
          ],
          dealer: false,
          acting: false,
          chipStackAmount: 800,
          winnings: 0,
          contribution: 600
        }
      ];
      const patient = (<AcpcTable
        boardCards={boardCards}
        potChips={1000}
        user={user}
        opponents={opponents}/>);
      const wrapper = shallow(patient);
      expect(renderer.create(patient).toJSON()).toMatchSnapshot();
    });
  });
});
