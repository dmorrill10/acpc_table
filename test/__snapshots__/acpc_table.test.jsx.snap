exports[`AcpcTable with 2-player Texas hold'em table on the river works 1`] = `
<div
  className="acpc-table">
  <ol
    className="opponents">
    <li>
      <ol
        className="player">
        <li
          className="held_player_information">
          <div
            className="name">
            player2
          </div>
          <ol
            className="card-list">
            <li>
              <div
                className="card">
                <div
                  className="logo" />
              </div>
            </li>
            <li>
              <div
                className="card">
                <div
                  className="logo" />
              </div>
            </li>
          </ol>
          <div
            className="player_stack">
            <div
              className="chip-stack">
              <div
                className="amount">
                800
              </div>
            </div>
          </div>
        </li>
        <li
          className="outer_player_information">
          <div
            className="contribution">
            <div
              className="chip-stack">
              <div
                className="amount">
                600
              </div>
            </div>
          </div>
        </li>
      </ol>
    </li>
  </ol>
  <div
    className="community-information">
    <ol
      className="board-cards">
      <li>
        <div
          className="card-face-up-h">
          <div
            className="top-left-rank">
            2
          </div>
          <div
            className="suit">
            ♥
          </div>
          <div
            className="bottom-right-rank">
            2
          </div>
        </div>
      </li>
      <li>
        <div
          className="card-face-up-d">
          <div
            className="top-left-rank">
            3
          </div>
          <div
            className="suit">
            ♦
          </div>
          <div
            className="bottom-right-rank">
            3
          </div>
        </div>
      </li>
      <li>
        <div
          className="card-face-up-s">
          <div
            className="top-left-rank">
            10
          </div>
          <div
            className="suit">
            ♠
          </div>
          <div
            className="bottom-right-rank">
            10
          </div>
        </div>
      </li>
      <li>
        <div
          className="card-face-up-c">
          <div
            className="top-left-rank">
            A
          </div>
          <div
            className="suit">
            ♣
          </div>
          <div
            className="bottom-right-rank">
            A
          </div>
        </div>
      </li>
      <li>
        <div
          className="card-face-up-d">
          <div
            className="top-left-rank">
            K
          </div>
          <div
            className="suit">
            ♦
          </div>
          <div
            className="bottom-right-rank">
            K
          </div>
        </div>
      </li>
    </ol>
    <div
      className="chip-stack">
      <div
        className="amount">
        1000
      </div>
    </div>
  </div>
  <ol
    className="player acting user">
    <li
      className="outer_player_information">
      <div
        className="dealer_button">
        Dealer
      </div>
      <div
        className="contribution">
        <div
          className="chip-stack">
          <div
            className="amount">
            400
          </div>
        </div>
      </div>
    </li>
    <li
      className="held_player_information">
      <div
        className="name">
        player1
      </div>
      <ol
        className="card-list">
        <li>
          <div
            className="card">
            <div
              className="logo" />
          </div>
        </li>
        <li>
          <div
            className="card">
            <div
              className="logo" />
          </div>
        </li>
      </ol>
      <div
        className="player_stack">
        <div
          className="chip-stack">
          <div
            className="amount">
            1000
          </div>
        </div>
      </div>
    </li>
  </ol>
</div>
`;
